strongbox-data-service/src/main/java/org/carlspring/strongbox/gremlin/repositories/GremlinRepository.java

```diff
@@ -0,0 +1,137 @@
+package org.carlspring.strongbox.gremlin.repositories;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import org.apache.tinkerpop.gremlin.structure.Element;
+import org.apache.tinkerpop.gremlin.structure.Graph;
+import org.carlspring.strongbox.data.domain.DomainObject;
+import org.carlspring.strongbox.gremlin.adapters.EntityTraversalAdapter;
+import org.carlspring.strongbox.gremlin.dsl.EntityTraversal;
+import org.carlspring.strongbox.gremlin.dsl.EntityTraversalSource;
+import org.carlspring.strongbox.gremlin.tx.TransactionContext;
+import org.neo4j.ogm.session.Session;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.repository.CrudRepository;
+
+/**
+ * Base implementation for Gremlin repositories.
+ *
+ * @author sbespalov
+ */
+@Transactional
+public abstract class GremlinRepository<S extends Element, E extends DomainObject> implements CrudRepository<E, String>
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(GremlinRepository.class);
+
+    @Inject
+    @TransactionContext
+    private Graph graph;
+    @Inject
+    protected Session session;
+
+    protected EntityTraversalSource g()
+    {
+        return graph.traversal(EntityTraversalSource.class);
+    }
+
+    protected abstract EntityTraversal<S, S> start(E entity, Supplier<EntityTraversalSource> g);
+
+    protected abstract EntityTraversal<S, S> start(Supplier<EntityTraversalSource> g);
+
+    public <R extends E> R save(Supplier<EntityTraversalSource> g, R entity) {
```

Comment:
Cosmetic: Braces on new lines please. :)
```suggestion
    public <R extends E> R save(Supplier<EntityTraversalSource> g, R entity)
    {
```

Reply:
fixed